index.js

```diff
@@ -32,54 +33,105 @@ load.path = function (dir) {
   var debug = getFirst(path.join(dir, 'build/Debug'), matchBuild)
   if (debug) return debug
 
-  var names = [platform + '-' + arch]
-  if (libc) names.push(platform + libc + '-' + arch)
-
-  if ((arch === 'arm' || arch === 'arm64') && armv) {
-    names.forEach(function (name) {
-      names.push(name + '-v' + armv)
-    })
-  }
-
   // Find most specific flavor first
-  for (var i = names.length; i--;) {
-    var prebuild = getFirst(path.join(dir, 'prebuilds/' + names[i]), matchPrebuild)
-    if (prebuild) return prebuild
-
-    var napiRuntime = getFirst(path.join(dir, 'prebuilds/' + names[i]), matchNapiRuntime)
-    if (napiRuntime) return napiRuntime
-
-    var napi = getFirst(path.join(dir, 'prebuilds/' + names[i]), matchNapi)
-    if (napi) return napi
-  }
-
-  throw new Error('No native build was found for runtime=' + runtime + ' abi=' + abi + ' platform=' + platform + libc + ' arch=' + arch)
+  var prebuilds = path.join(dir, 'prebuilds', platform + '-' + arch)
+  var parsed = readdirSync(prebuilds).map(parseTags)
+  var candidates = parsed.filter(matchTags(runtime, abi))
+  var winner = candidates.sort(compareTags(runtime))[0]
+  if (winner) return path.join(prebuilds, winner.file)
+
+  var target = [
+    'platform=' + platform,
+    'arch=' + arch,
+    'runtime=' + runtime,
+    'abi=' + abi,
+    'uv=' + uv,
+    armv ? 'armv=' + armv : '',
+    'libc=' + libc
+  ].filter(Boolean).join(' ')
+
+  throw new Error('No native build was found for ' + target)
 }
 
-function getFirst (dir, filter) {
+function readdirSync (dir) {
   try {
-    var files = fs.readdirSync(dir).filter(filter)
-    return files[0] && path.join(dir, files[0])
+    return fs.readdirSync(dir)
   } catch (err) {
-    return null
+    return []
   }
 }
 
-function matchNapiRuntime (name) {
-  return name === runtime + '-napi.node'
+function getFirst (dir, filter) {
+  var files = readdirSync(dir).filter(filter)
+  return files[0] && path.join(dir, files[0])
 }
 
-function matchNapi (name) {
-  return name === 'node-napi.node'
+function matchBuild (name) {
+  return /\.node$/.test(name)
 }
 
-function matchPrebuild (name) {
-  var parts = name.split('-')
-  return parts[0] === runtime && parts[1] === abi + '.node'
+function parseTags (file) {
+  var arr = file.split('.')
+  var extension = arr.pop()
+  var tags = { file: file, specificity: 0 }
+
+  if (extension !== 'node') return
+
+  for (var i = 0; i < arr.length; i++) {
+    var tag = arr[i]
+
+    if (tag === 'node' || tag === 'electron' || tag === 'node-webkit') {
+      tags.runtime = tag
+    } else if (tag === 'napi') {
+      tags.napi = true
+    } else if (tag.slice(0, 3) === 'abi') {
+      tags.abi = tag.slice(3)
+    } else if (tag.slice(0, 2) === 'uv') {
+      tags.uv = tag.slice(2)
+    } else if (tag.slice(0, 4) === 'armv') {
+      tags.armv = tag.slice(4)
+    } else if (tag === 'glibc' || tag === 'musl') {
+      tags.libc = tag
+    } else {
+      return
```

Comment:
Shouldn’t this be continue?

Reply:
I figured if there's a filename containing an unknown tag, the file should be ignored altogether. No?

Reply:
I think it should just ignore the tag. Sounds more future proof to me